### ISSUE TITLE
--root-dir no longer seems to work

### ISSUE THREAD

**Issue by anlutro** [netz98/n98-magerun2#224]
The following command works fine with the original magerun:

```
n98-magerun.phar sys:check --root-dir=/path/to/magento
```

However, the exact same command fails with magerun2, with this exception message:

```
  [RuntimeException]
  Magento folder could not be detected
```

Versions:

```
n98-magerun version 1.97.22
n98-magerun2 version 1.1.16
```

**Comment by tkn98** [comment]
Please put the `--root-dir` switch at the beginning, before the command name (`sys:check`), as it is a global parameter:

    n98-magerun2.phar --root-dir=/path/to/magento sys:check

This should solve the Magento detection issue.

**Comment by anlutro** [comment]
That's pretty annoying. Symfony's console component allows arbitrarily placed command line options as far as I know, I thought you were using that? Furthermore, symfony console applications will usually warn you if you provide wrong arguments:

```
% composer about --fasdkj
  [Symfony\Component\Console\Exception\RuntimeException]  
  The "--fasdkj" option does not exist.                   
```

**Comment by tkn98** [comment]
I do see the issue that this is less stable. 
Might be the version we use.
For the warning: As the option exists, it's not that warning.

For a better usability, this needs further review.

Another usability issue is that the =" is not POSIX conform, so 

    n98-magerun2.phar --root-dir /path/to/magento sys:check

should work as well. Just noting, have not checked it so far.

**Comment by ktomk** [comment]
We believe that the bug reported is fixed in the development version. It can be upgraded to it using the `self-update` command with the `--unstable` switch.

**Comment by ktomk** [comment]
Background: Symfony console is used to define the option and it's argument state, but at the stage we parse the argument, we used `getopt()` which stopped at the first operand argument which is the Symfony command (e.g. "`sys:check`"). I now switched to a new parser which does not stop scanning for the option on operands but at the end of options ("`--`").

**Comment by ktomk** [comment]
We believe that next to "root-dir", the same fix for the other three
options "skip-root-check", "skip-config" and "skip-core-commands" are
fixed in the development version, too.

Thanks again for your report, I'll close the issue for now. Please 
raise any concerns or feedback you have and/or if you need a tagged
release soon.

**Comment by tkn98** [comment]
This has been shipped with Magerun2 1.1.17
